Helper/ShipmentOptions.php

```diff
@@ -0,0 +1,323 @@
+        string $entityId
+    ): ?string {
+        $sql = $connection
+            ->select()
+            ->from($tableName, ['value'])
+            ->where('attribute_id = ?', $attributeId)
+            ->where('entity_id = ?', $entityId);
+
+        return $connection->fetchOne($sql);
+    }
+
+    /**
+     * @return bool
+     */
+    public function hasLargeFormat(): bool
+    {
+        return self::$defaultOptions->getDefaultLargeFormat(self::LARGE_FORMAT);
+    }
+
+    /**
+     * @return string
+     */
+    public function getLabelDescription(): string
+    {
+        $checkoutData     = $this->order->getData('myparcel_delivery_options');
+        $labelDescription = $this->helper->getGeneralConfig(
+            'print/label_description',
+            $this->order->getStoreId()
+        );
+
+        if (! $labelDescription) {
+            return '';
+        }
+
+        $productInfo      = $this->getItemsCollectionByShipmentId($this->order->getId());
+        $deliveryDate     = date('d-m-Y', strtotime($this->helper->convertDeliveryDate($checkoutData)));
+        $labelDescription = str_replace(
+            [
+                self::ORDER_NUMBER,
+                self::DELIVERY_DATE,
+                self::PRODUCT_ID,
+                self::PRODUCT_NAME,
+                self::PRODUCT_QTY,
+            ],
+            [
+                $this->order->getIncrementId(),
+                $this->helper->convertDeliveryDate($checkoutData) ? $deliveryDate : '',
+                $this->getProductInfo($productInfo, 'product_id'),
+                $this->getProductInfo($productInfo, 'name'),
+                round($this->getProductInfo($productInfo, 'qty'), 0),
+            ],
+            $labelDescription
+        );
+
+        return (string) $labelDescription;
+    }
+
+    /**
+     * @param  array  $productInfo
+     * @param  string $field
+     *
+     * @return string|null
+     */
+    private function getProductInfo(array $productInfo, string $field): ?string
+    {
+        if ($productInfo) {
+            return $productInfo[0][$field];
+        }
+
+        return null;
+    }
+
+    /**
+     * @param $shipmentId
+     *
+     * @return array
+     */
+    public function getItemsCollectionByShipmentId($shipmentId): array
+    {
+        /** @var \Magento\Framework\App\ResourceConnection $connection */
+        $connection = $this->objectManager->create(ResourceConnection::class);
+        $conn       = $connection->getConnection();
+        $select     = $conn->select()
+            ->from(
+                ['main_table' => $connection->getTableName('sales_shipment_item')]
+            )
+            ->where('main_table.parent_id=?', $shipmentId);
+        return $conn->fetchAll($select);
+    }
+
+    /**
+     * Get default value if option === null
+     *
+     * @param      $optionKey
+     *
+     * @return bool
+     * @internal param $option
+     */
+    private function getValueOfOption($optionKey): bool
```

Comment:
En die default is ook altijd een boolean? Moet je hem dan niet `optionIsEnabled` oid noemen?